internal/artifacts/module_arch.go

```diff
@@ -76,11 +70,32 @@ func ExecuteSoloBuild(source, target string, extensions []string, modulesNames [
 		return errors.Wrapf(err, getBuildErrMsg(buildFailedMsg, multiBuildFailedMsg, modulesNames))
 	}
 
-	selectedModulesMap := convertListToMap(modulesNames)
+	selectedModulesMap := make(map[string]bool)
+	var selectedModulesWithDependenciesMap map[string]bool
 
-	sortedModules := sortSelectedModules(allModulesSorted, selectedModulesMap)
+	for _, moduleName := range modulesNames {
+		selectedModulesMap[moduleName] = true
+	}
 
-	err = buildSelectedModules(sourceDir, target, extensions, mtaObj, sortedModules, allDependencies, wdGetter)
+	if allDependencies {
+		selectedModulesWithDependenciesMap = make(map[string]bool)
+		for module := range selectedModulesMap {
+			err = collectSelectedModulesAndDependencies(mtaObj, selectedModulesWithDependenciesMap, module)
+			if err != nil {
+				return errors.Wrapf(err, getBuildErrMsg(buildFailedMsg, multiBuildFailedMsg, modulesNames))
+			}
+		}
+	} else {
+		selectedModulesWithDependenciesMap = selectedModulesMap
+	}
+
+	if len(selectedModulesWithDependenciesMap) > len(selectedModulesMap) {
```

Comment:
I'm not sure if this is the condition that should be checked to decide which message should be used. Maybe just check `allDependencies`?

Reply:
I wanted to emphasize that set of modules is extended in case when additional modules are added. If you think it has no sense I'll simplify this.

Reply:
I think the user only cares about the parameter they sent and not if additional modules will actually be built or not (they can see this in the modules list anyway). I think it would be confusing when there are no dependencies, the user wouldn't be sure if the `allDependencies` parameter is taken into account or not.

Reply:
in case of no dependencies still we change an order and I think we want to list it, not?

Reply:
We list the modules in the order they will be built anyway, the difference is only in the error message

Reply:
done